lib/flapjack/gateways/web.rb

```diff
@@ -301,6 +299,14 @@ def logger
         @check_details              = entity_check.details
         @check_perfdata             = entity_check.perfdata
 
+        @check_initial_failure_delay = entity_check.initial_failure_delay ||
+          Flapjack::DEFAULT_INITIAL_FAILURE_DELAY
+        @check_repeat_failure_delay  = entity_check.repeat_failure_delay  ||
+          Flapjack::DEFAULT_REPEAT_FAILURE_DELAY
+
+        @check_initial_failure_delay_is_default = entity_check.initial_failure_delay ? false : true
```

Comment:
Could be `entity_check.initial_failure_delay.nil?`, and similarly for the other one.